### PR TITLE
Log diffs on status changes, select correct mopping operation and add fan support for Viomi

### PR DESCRIPTION
**Pull request by rumpeltux**

### CODE REVIEW

File: lib/devices/Viomi.js [hunk 1]
```diff
@@ -48,10 +64,16 @@ class Viomi extends MiioVacuum {
...
          */
         this.operation = [0, 1, 0];
 
-        /** @type {number} Current operation mode. Probably equal to MopMode with 3=zoned clean. */
+        /**
+         * @type {number} Current operation mode.
+         * 0="normal cleaning"
+         * 1="mop moves" (unsure if only enabled in mop-mode)
+         */
         this.mode = 0;
```
Comment (Hypfer) [inline lib/devices/Viomi.js:77]: While this isn't even part of the PR, this should be refactored to an enum object and referenced by name.

e.g. `MODES.NORMAL_CLEANING`

Not in this PR though

**Comment by Hypfer** [comment]
lgtm

File: lib/devices/Viomi.js (hunk 1 above)
Comment (rumpeltux) [reply lib/devices/Viomi.js]: Done.

File: lib/devices/Viomi.js [hunk 2]
```diff
@@ -48,11 +71,13 @@ class Viomi extends MiioVacuum {
         /**
          * Current cleaning operation.
          * [mode, pause/start (, opt: roomspec)]
+         *
+         * mode reflects this.mode
          */
         this.operation = [0, 1, 0];
```
Comment (Hypfer) [inline lib/devices/Viomi.js:77]: You've missed this mode reference

File: lib/devices/Viomi.js [hunk 3]
```diff
@@ -136,10 +162,23 @@ class Viomi extends MiioVacuum {
...
+    /** @override */
+    onStatusChange(changedProperties) {
+        if (changedProperties.includes("box_type") || changedProperties.includes("mop_type")) {
             this.checkMopMode();
         }
+        if (changedProperties.includes("state")) {
+            if (["CHARGING", "RETURNING_HOME"].includes(this.status.state)){
+                // The pending operation completed. Reset, so that subsequence startCleaning
+                // calls can properly start a full clean.
+                this.operation = [0, 1, 0];
```
Comment (Hypfer) [inline lib/devices/Viomi.js:179]: And this one.

That's all I can see in this diff view

**Pull request merged**